stackv2/to_dolma.py

```diff
@@ -0,0 +1,63 @@
+import argparse
+import os
+from datetime import datetime
+
+from datasets import load_dataset
+
+from licensed_pile.licenses import PermissiveLicenses
+from licensed_pile.logs import configure_logging
+from licensed_pile.write import to_dolma
+
+SOURCE_NAME = "The Stack v2"
+
+logger = configure_logging("court-listener-opinion")
+
+LICENSE_MAP = {
+    "Apache-2.0": PermissiveLicenses.APACHE_2,
+    "BSD-2-Clause": PermissiveLicenses.BSD,
+    "MIT License": PermissiveLicenses.MIT,
+}
+
+
+def main(args):
+    ds = load_dataset(args.data_path, data_dir=args.data_dir, split="train")
+    ds.filter(lambda x: x["license"] in LICENSE_MAP.keys())
+    ds = ds.map(
+        lambda x: {
+            "id": x["id"],
+            "text": x["content"],
+            "source": SOURCE_NAME,
+            "added": datetime.utcnow().isoformat(),
+            "created": "2024-02-29",
+            "metadata": {
+                "license": str(LICENSE_MAP[x["license"]]),
+                "url": "https://github.com/" + x["repo"] + "/tree/" + x["revision_id"],
+            },
+        }
+    )
+
+    output_file_base_name = os.path.join(args.output_dir, "starcoder2data.jsonl.gz")
```

Comment:
You don't need to join the output_dir and `"startcoder2data.jsonl.gz"`, you should just pass the latter.